Bottom: 40a4c6e627833ef07fcb957a495e297a004f643a
Top:    88e448b44788f727edff038848d6a8a700cf4797
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-25 16:09:13 +0100

Refresh of move-drawplainpin-and-drawplai

---

diff --git a/src/draw.c b/src/draw.c
index 8b73644..1fdd9c4 100644
--- a/src/draw.c
+++ b/src/draw.c
@@ -84,8 +84,6 @@ static void DrawEverything (BoxTypePtr);
 static void DrawPPV (int group, const BoxType *);
 static int DrawLayerGroup (int, const BoxType *);
 static void DrawPinOrViaLowLevel (PinTypePtr, bool);
-static void DrawPlainPin (PinTypePtr, bool);
-static void DrawPlainVia (PinTypePtr, bool);
 static void DrawPinOrViaNameLowLevel (PinTypePtr);
 static void DrawPadLowLevel (hidGC, PadTypePtr, bool, bool);
 static void DrawPadNameLowLevel (PadTypePtr);
@@ -460,14 +458,20 @@ static int
 via_callback (const BoxType * b, void *cl)
 {
   PinTypePtr via = (PinTypePtr) b;
-  DrawPlainVia (via, false);
+  SetPVColor (Via, VIA_TYPE);
+  DrawPinOrViaLowLevel (Via, false);
+  if (!TEST_FLAG (HOLEFLAG, Via) && TEST_FLAG (DISPLAYNAMEFLAG, Via))
+    DrawPinOrViaNameLowLevel (Via);
   return 1;
 }
 
 static int
 pin_callback (const BoxType * b, void *cl)
 {
-  DrawPlainPin ((PinTypePtr) b, false);
+  SetPVColor (Pin, PIN_TYPE);
+  DrawPinOrViaLowLevel (Pin, holeToo);
+  if (!TEST_FLAG (HOLEFLAG, Pin) && TEST_FLAG (DISPLAYNAMEFLAG, Pin))
+    DrawPinOrViaNameLowLevel (Pin);
   return 1;
 }
 
@@ -1180,19 +1184,6 @@ DrawVia (PinTypePtr Via)
 }
 
 /* ---------------------------------------------------------------------------
- * draw a via without dealing with polygon clearance 
- */
-static void
-DrawPlainVia (PinTypePtr Via, bool holeToo)
-{
-  if (!Gathering)
-    SetPVColor (Via, VIA_TYPE);
-  DrawPinOrViaLowLevel (Via, holeToo);
-  if (!TEST_FLAG (HOLEFLAG, Via) && TEST_FLAG (DISPLAYNAMEFLAG, Via))
-    DrawPinOrViaNameLowLevel (Via);
-}
-
-/* ---------------------------------------------------------------------------
  * draws the name of a via
  */
 void
@@ -1225,19 +1216,6 @@ DrawPin (PinTypePtr Pin)
 }
 
 /* ---------------------------------------------------------------------------
- * draw a pin without clearing around polygons 
- */
-static void
-DrawPlainPin (PinTypePtr Pin, bool holeToo)
-{
-  if (!Gathering)
-    SetPVColor (Pin, PIN_TYPE);
-  DrawPinOrViaLowLevel (Pin, holeToo);
-  if (!TEST_FLAG (HOLEFLAG, Pin) && TEST_FLAG (DISPLAYNAMEFLAG, Pin))
-    DrawPinOrViaNameLowLevel (Pin);
-}
-
-/* ---------------------------------------------------------------------------
  * draws the name of a pin
  */
 void
